middlewares/authentication.js

```diff
@@ -51,33 +51,30 @@ const _getAccessToken = async (ctx) => {
           ctx.throw(500, JSON.stringify({error: {status: 500, error_message: e}}));
         }
       }
-
       ctx.state.accessToken = user.access_token;
       return user;
     })
     .catch( err => {
-      ctx.throw(401, 'unauthorized');
+      ctx.throw(401, 'unauthorized sucka', err);
     });
-
   ctx.status = 200;
-  ctx.body = responseBody;
+  ctx.body = UserSerializer.serializeWithToken(responseBody);
 }
 
 async function checkAuth (ctx, next) {
-  console.log('check for params code and header.authorization');
-  if (ctx.params && ctx.params.code) {
-    _getAccessToken(ctx);
+  if (ctx.query && ctx.query.code) {
+    await _getAccessToken(ctx);
   } else if (!ctx.header.authorization) {
     ctx.throw(401, 'unauthorized');
   }
-  // made it through the gaunlet, check the access_token
-  const accessToken = ctx.header.authorization.split(' ')[1];
-  const user = User.find({access_token: accessToken});
-  if (!user['access_token']) {
-    // not a valid access_token
-    ctx.throw(401, 'unauthorized');
+  if (!ctx.state.accessToken) {
+    const user = User.find({access_token: accessToken});
```

Comment:
I solved some linter errors, but I don't know how to solve this one? Did you want to take the `accessToken` from the headers?